server/protocols/webadmin/cli/james_cli

```diff
@@ -0,0 +1,209 @@
+
+    res = req.put(ctx.obj["path"] + "/domains/" + domain_name)
+
+    if res.status_code == 204:
+        print("%s has been added." % domain_name)
+    elif res.status_code == 400:
+        print("Invalid request for domain creation.")
+    elif res.status_code == 401:
+        print("Authentication error.")
+    elif res.status_code == 500:
+        print("Internal server error!")
+    else:
+        print("Unhandled response received from the server.")
+        print("Status: " + str(res.status_code))
+        print_json(res)
+
+
+@domain.command(name="remove")
+@click.pass_context
+@click.argument("domain_name", type=str)
+def delete_domain(ctx, domain_name):
+    """Remove a domain from the domain list"""
+    res = req.delete(ctx.obj["path"] + "/domains/" + domain_name)
+
+    if res.status_code == 204:
+        print("%s has been removed." % domain_name)
+    elif res.status_code == 401:
+        print("Authentication error.")
+    elif res.status_code == 500:
+        print("Internal server error!")
+    else:
+        print("Unhandled response received from the server.")
+        print("Status: " + str(res.status_code))
+        print_json(res)
+
+
+@domain.command(name="exist")
+@click.pass_context
+@click.argument("domain_name", type=str)
+def check_existence(ctx, domain_name):
+    """Check whether a domain is existing or not"""
+    res = req.get(ctx.obj["path"] + "/domains/" + domain_name)
+    if res.status_code == 204:
+        print("%s does exist." % domain_name)
+    elif res.status_code == 401:
+        print("Authentication error.")
+    elif res.status_code == 404:
+        print("%s does NOT exist." % domain_name)
+    elif res.status_code == 500:
+        print("Internal server error!")
+    else:
+        print("Unhandled response received from the server.")
+        print("Status: " + str(res.status_code))
+        print_json(res)
+
+
+@main.group()
+def domain_alias():
+    """Domain's alias implementation"""
+    pass
+
+
+@domain_alias.command(name="list")
+@click.pass_context
+@click.argument("domain_name", type=str)
+def get_domain_aliases(ctx, domain_name):
+    """Show all aliases of a domain"""
+    res = req.get(ctx.obj["path"] + "/domains/" + domain_name + "/aliases")
+    if res.status_code == 200:
+        print_json(res)
+    elif res.status_code == 401:
+        print("Authentication error.")
+    elif res.status_code == 404:
+        print("The domain does not exist.")
+    elif res.status_code == 500:
+        print("Internal server error!")
+    else:
+        print("Unhandled response received from the server.")
+        print("Status: " + str(res.status_code))
+        print_json(res)
+
+
+@domain_alias.command(name="add")
+@click.pass_context
+@click.argument("source_domain", type=str)
+@click.argument("destination_domain", type=str)
+def add_alias(ctx, source_domain, destination_domain):
+    """Add an alias to a destination domain"""
+    req.put(ctx.obj["path"] + "/domains/" + source_domain)
```

Comment:
IMO we should not attempt to create a domain when creating a domainAlias

We shoiuld rather tell the user `We cannot create the alias becaus XXX does not exist. Please create it first.`